### PR TITLE
repeat last yaw and pitch if line section is '*'

### PR DESCRIPTION
**Pull request by philippTheCat**

### CODE REVIEW

**Comment by YaLTeR** [comment]
Eh I dunno about this tbh.

**Comment by YaLTeR** [comment]
Although I suppose it can be helpful. Do you have any other suggestions for the actual character? Not sure `*` is the best fit here.

EDIT: how about `_` which is used in some languages as a placeholder.

File: src/hltas.cpp [hunk 1]
```diff
@@ -372,6 +372,10 @@ namespace HLTAS
 		bool yawIsRequired = false; // For viewangles checking.
 		int strafeDir = -1; // For viewangles checking.
 		std::string line;
+
+		float lastYaw = 0, lastPitch = 0;
+
+
```
Comment (YaLTeR) [inline src/hltas.cpp:378]: Extra blank line

File: src/hltas.cpp [hunk 2]
```diff
@@ -372,6 +372,10 @@ namespace HLTAS
 		bool yawIsRequired = false; // For viewangles checking.
 		int strafeDir = -1; // For viewangles checking.
 		std::string line;
+
+		float lastYaw = 0, lastPitch = 0;
```
Comment (YaLTeR) [inline src/hltas.cpp:376]: `lastYaw` should be `double`.

File: src/hltas.cpp [hunk 3]
```diff
@@ -597,7 +601,7 @@ namespace HLTAS
 				{
 					if (l == 0)
 						throw ErrorCode::FAILFRAME;
-					if (!std::isdigit(str[0]) && str[0] != '-')
+					if (!std::isdigit(str[0]) && (str[0] != '-' && str[0] != '*'))
```
Comment (YaLTeR) [inline src/hltas.cpp:604]: Extra `()` not needed.

File: src/hltas.cpp [hunk 4]
```diff
@@ -614,34 +618,47 @@ namespace HLTAS
 						char *s2;
 						f.X = std::strtod(s, &s2); // TODO: replace this with lexical_cast probably.
 						f.Y = std::strtod(s2 + 1, nullptr);
+					} else if(str == "*") {
+						f.Yaw = lastYaw;
```
Comment (YaLTeR) [inline src/hltas.cpp:622]: Well, what happens if there's no last yaw? That's right, undefined behavior, which is bad. Make `lastYaw` a `std::optional` and make it an error if it's not present.

File: src/hltas.cpp [hunk 5]
```diff
@@ -614,34 +618,47 @@ namespace HLTAS
...
 					} else {
-						try {
-							f.Yaw = boost::lexical_cast<double>(str);
-						} catch (boost::bad_lexical_cast) {
-							throw ErrorCode::FAILFRAME;
+							try {
```
Comment (YaLTeR) [inline src/hltas.cpp:624]: You messed up the indentation on this block up until `break;`. It should be one level to the left.

File: src/hltas.cpp [hunk 6]
```diff
@@ -614,34 +618,47 @@ namespace HLTAS
...
 					}
-				}
 					break;
 
 				case 5:
 				{
 					if (l == 0)
 						throw ErrorCode::FAILFRAME;
-					if (!std::isdigit(str[0]) && str[0] != '-')
+					if (!std::isdigit(str[0]) && (str[0] != '-' && str[0] != '*'))
```
Comment (YaLTeR) [inline src/hltas.cpp:639]: Extra `()` not needed.

File: src/hltas.cpp [hunk 7]
```diff
@@ -614,34 +618,47 @@ namespace HLTAS
...
 						break;
 
 					f.PitchPresent = true;
-					try {
-						f.Pitch = boost::lexical_cast<float>(str);
-					} catch (boost::bad_lexical_cast) {
-						throw ErrorCode::FAILFRAME;
+
+					if (str == "*") {
+						f.Pitch = lastPitch;
```
Comment (YaLTeR) [inline src/hltas.cpp:648]: Same thing about undefined behaviour.

File: src/hltas.cpp [hunk 8]
```diff
@@ -614,34 +618,47 @@ namespace HLTAS
...
+					}
+					else {
```
Comment (YaLTeR) [inline src/hltas.cpp:650]: Should be on the previous line.

File: src/hltas.cpp [hunk 9]
```diff
@@ -614,34 +618,47 @@ namespace HLTAS
...
+
```
Comment (YaLTeR) [inline src/hltas.cpp:651]: Extra blank line.

File: src/hltas.cpp [hunk 10]
```diff
@@ -614,34 +618,47 @@ namespace HLTAS
...
+						try {
+							f.Pitch = boost::lexical_cast<float>(str);
+							lastPitch = f.Pitch;
+						}
+						catch (boost::bad_lexical_cast) {
```
Comment (YaLTeR) [inline src/hltas.cpp:656]: Should be on the previous line.

File: src/hltas.cpp [hunk 11]
```diff
@@ -614,34 +618,47 @@ namespace HLTAS
...
+							throw ErrorCode::FAILFRAME;
+						}
 					}
 				}
-					break;
+				break;
```
Comment (YaLTeR) [inline src/hltas.cpp:661]: Should be one indentation to the right.

**Review by YaLTeR — CHANGES_REQUESTED** [review]
Here's a review.

**Comment by Matherunner** [comment]
Maybe the ^ character instead of *?

**Comment by YaLTeR** [comment]
Hm that could work yeah

**Comment by YaLTeR** [comment]
This PR is now stale as hltas was rewritten from scratch.

**Pull request closed without merging**